diff --git a/node_modules/expo-image/android/build.gradle b/node_modules/expo-image/android/build.gradle
index d8ca14a..43c9866 100644
--- a/node_modules/expo-image/android/build.gradle
+++ b/node_modules/expo-image/android/build.gradle
@@ -14,6 +14,9 @@ buildscript {
   }
 }
 
+// Ustawiamy flagę, aby wykluczyć ExpoImageAppGlideModule z kompilacji
+ext.excludeAppGlideModule = true
+
 android {
   def rnVersion = getRNVersion()
   namespace "expo.modules.image"
@@ -25,10 +28,14 @@ android {
     } else {
       consumerProguardFiles('proguard-rules.pro')
     }
-
     buildConfigField("boolean", "ALLOW_GLIDE_LOGS", project.properties.get("EXPO_ALLOW_GLIDE_LOGS", "false"))
   }
 
+  // W razie duplikacji klasy GeneratedAppGlideModuleImpl – wybieramy pierwszą napotkaną wersję
+  packagingOptions {
+      pickFirst '**/GeneratedAppGlideModuleImpl.class'
+  }
+
   sourceSets {
     main {
       java {
@@ -45,8 +52,12 @@ dependencies {
 
   implementation 'com.facebook.react:react-android'
 
-  api "com.github.bumptech.glide:glide:${GLIDE_VERSION}"
-  kapt "com.github.bumptech.glide:compiler:${GLIDE_VERSION}"
+  api("com.github.bumptech.glide:glide:${GLIDE_VERSION}") {
+    exclude group: 'com.github.bumptech.glide'
+  }
+  // Nie stosujemy bloku exclude przy kapt – wystarczy zwykła deklaracja:
+  // kapt "com.github.bumptech.glide:compiler:${GLIDE_VERSION}"
+  
   api 'com.caverock:androidsvg-aar:1.4'
 
   implementation "com.github.penfeizhou.android.animation:glide-plugin:3.0.1"
@@ -69,15 +80,9 @@ def getNodeModulesPackageVersion(packageName, overridePropName) {
     commandLine("node", "-e", "console.log(require('$packageName/package.json').version);")
   }.standardOutput.asText.get().trim()
   def version = safeExtGet(overridePropName, nodeModulesVersion)
-
   def coreVersion = version.split("-")[0]
   def (major, minor, patch) = coreVersion.tokenize('.').collect { it.toInteger() }
-
-  return versionToNumber(
-      major,
-      minor,
-      patch
-  )
+  return versionToNumber(major, minor, patch)
 }
 
 def getRNVersion() {
